vignettes/data_structures.Rmd

```diff
@@ -0,0 +1,122 @@
+---
+title: "Data Structures"
+author: "Lorenz Walthert and Kirill Müller"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Vignette Title}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+This vignette illustrates how the core of `styler` currently^[at commit e`6ddee0f510d3c9e3e22ef68586068fa5c6bc140`] works, i.e. how
+rules are applied to a parse table and how limitations of this approach can be 
+overcome with a refined approach.
+
+## Status quo - the flat apporach
+
+Roughly speaking, a string containing code to be formatted is parsed with `parse`
+and the output is passed to `getParseData` in order to obtain a parse
+table with detailed information about every token. For a simple example string
+"`a <- function(x) { 1+1 }`" to be formatted, the parse table on which `styler`
+performs the manipulations looks similar to the one presented below.
+```{r, message = FALSE}
+library("styler")
+library("dplyr")
+
+code <- "a <- function(x) { 1+1 }"
```

Comment:
Perhaps you could use a function that has another brace expression (perhaps with an `if` clause), and comment on the structure of both brace subexpressions in the nested parse trees?